content/glossary/Glossary.md

```diff
@@ -1,145 +1,345 @@
+
+---
 
 ## Contrast
 
-**Contrast** is a new security technology for finding vulnerabilities in an application. It finds vulnerabilities by instrumenting the application and virtual machine with sensors and watching the stream of generated events for patterns that represent vulnerabilities. The term **Contrast** encompasses the entire suite of technologies around this product, including the *JVM plugin* (the *engine*), and the Contrast site (the *TeamServer*).
+Contrast is a security technology to **Assess** your **application** for vulnerabilities and **Protect** it from **attacks**. Contrast finds vulnerabilities by instrumenting the application and virtual machine with sensors, and watching the stream of generated events for patterns that represent vulnerabilities. The term encompasses the entire suite of technologies around this product, including the **agents** and **Contrast UI**.
 
 ---
 
-## Contrast JVM Plugin (Agent)
+## Contrast Service
 
-The **Contrast JVM Plugin** is a Java agent that is installed in the application server that is hosting the application(s) to be analyzed. This *engine* generates *traces* that contain vulnerability results detected by the *engine* while the application is operating. It periodically sends these *traces* to its associated *TeamServer* for further analysis by a human being.
+The Contrast Service is an executable written in Go that handles communication with the **Contrast UI** for the Ruby, Python and Proxy **agents**. For the Proxy agent, it also acts as an analysis **engine** that can apply Protect rules to models representing HTTP requests.
 
 ---
 
-## Contrast Site (TeamServer)
+## Contrast UI 
 
-The **Contrast site** is a web application that receives *traces* from Contrast *engines* and allows the analyst to view and triage the results.
+The **Contrast UI** is the web interface from which users download agents and view information on their **applications** and reported **vulnerabilities**. This also includes the REST API to which the **agents** report data.
 
 ---
 
 ## Creation Event
 
-When an *untrusted data source* provides data to the application, a **Creation Event** (or **Source Event**) is generated that documents what occurred. This information is stored with the *tracked* data object that is created and is used to describe where untrusted data came from if a *rule* generates a *trace*.
+When an **untrusted data source** provides data to an application, Contrast generates a creation event (or source event) that documents what occurred. Contrast stores this information with the **tracked** data object that it created, and uses it to describe the source of the untrusted data, if a **rule** generates a **trace**.
+
```

Comment:
@DanFiedler We don't use the term "term" anywhere else in the docs (or the UI, as far as I know). Can we update terms to "vulnerability"?